src/Festung/Vault/VaultHandler.hs

```diff
@@ -109,6 +112,10 @@ query o v q = flattenError <$> sendMappedCommand f v q
 
 
 -- | Execute a parametrized query
-parametrizedQuery :: VaultOpener -> Vault -> String -> [P.Value] -> IO (Either VaultError P.QueryResult)
-parametrizedQuery o v q p = flattenError <$> sendMappedCommand f v (q, p)
+parametrizedQuery :: VaultOpener -> Vault -> String -> P.QueryParams -> IO (Either VaultError P.QueryResult)
+parametrizedQuery o v q (P.Params p) = flattenError <$> sendMappedCommand f v (q, p)
     where f = (,) o . ParametrizedQuery
+parametrizedQuery o v q (P.BulkParams bp) = do
+    eithers <- mapM (parametrizedQuery o v q . P.Params) bp
+    let error = find isLeft eithers
+    maybe (return $ last eithers) return error
```

Comment:
I guess we somehow have to accumulate the value of `rowsChanged` of the `QueryResult`s here. See me other comment in the tests for more info.